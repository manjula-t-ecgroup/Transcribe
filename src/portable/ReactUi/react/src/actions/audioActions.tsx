import Axios from 'axios';
import { COPY_AUDIO, INITIAL_TRANSCRIPTION, JUMP_CHANGE, PLAY_STATUS, PLAYSPEEDRATE_CHANGE,
    REPORT_POSITION, REQUEST_POSITION, SAVE_STATUS, SAVE_TOTAL_SECONDS, SUBMIT_STATUS } from './types';


export function playStatus(playing: boolean): any{
    return {
        payload: playing,
        type: PLAY_STATUS
    }
}

export function playSpeedRateChange(playSpeedRate: number): any{
    return {
        payload: playSpeedRate,
        type: PLAYSPEEDRATE_CHANGE
    }
}

export function jumpChange(jump: number): any{
    return {
        payload: jump,
        type: JUMP_CHANGE
    }
}

export function saveTotalSeconds(seconds: number): any {
    return {
        payload: seconds,
        type: SAVE_TOTAL_SECONDS
    }
}

export const reportPosition = (taskid: string, seconds: number) => (dispatch: any) => {
    dispatch({payload: seconds, type: REPORT_POSITION});
    Axios.put('/api/ReportPosition?task=' + taskid + '&position=' + seconds.toString());
}

export function requestPosition(): any {
    return {
        type: REQUEST_POSITION
    }
}

export function setInitialTranscription(status: boolean): any {
    return {
        payload: status,
        type: INITIAL_TRANSCRIPTION
    }
}

export function setSubmitted(submit: boolean): any{
    return {
        payload: submit,
        type: SUBMIT_STATUS
    }
}

export function setSaved(saved: boolean): any{
    return {
        payload: saved,
        type: SAVE_STATUS
    }
}

export const copyAudio = (task: string, project: string, audioFile: string, data: object) => (dispatch: any) => {
    dispatch({type: COPY_AUDIO});
    Axios.put('/api/CopyAudio?task=' + task + '&project=' + project+ '&audioFile=' + audioFile, data)
}