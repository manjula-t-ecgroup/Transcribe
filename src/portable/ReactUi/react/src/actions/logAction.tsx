import Axios from 'axios';

export const log = (message: string) => {
    Axios.put('/api/Log?' + message);
}
