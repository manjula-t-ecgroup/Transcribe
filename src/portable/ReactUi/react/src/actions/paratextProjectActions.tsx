import Axios from 'axios';
import { FETCH_PARATEXT_PROJECTS, SELECT_PARATEXT_PROJECT } from './types';

export const fetchParatextProjects = () => (dispatch: any) => {
    // tslint:disable-next-line:no-console
    console.log("in get")
    Axios.get('/api/GetParatextProjects').
        then(paratextProjects => {
            dispatch({
                payload: paratextProjects,
                type: FETCH_PARATEXT_PROJECTS
            });
        });
}

export function selectParatextProject(id: string): any{
    return {
        payload: id,
        type: SELECT_PARATEXT_PROJECT
    }
}