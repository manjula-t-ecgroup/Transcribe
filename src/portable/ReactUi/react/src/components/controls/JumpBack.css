.JumpBack {
    color: rgba(0,0,0,0.6);
    display: flex;
    align-items: center;
    margin: 10px;
    font-size: 28pt;
}