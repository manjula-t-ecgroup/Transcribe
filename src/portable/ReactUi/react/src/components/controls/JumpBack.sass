@import "../../_variables.sass"

.JumpBack 
	color: rgba(0,0,0,0.6)
	display: flex
	align-items: center
	margin: 10px
	font-size: 28pt
	&:hover 
		-moz-border-radius: 5px
		border-radius: 5px
		-moz-box-shadow: 0px 0px 30px 0px $white
		box-shadow: 0px 0px 30px 0px $white