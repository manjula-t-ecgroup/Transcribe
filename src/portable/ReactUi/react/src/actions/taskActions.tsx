import Axios from 'axios';
import { setSubmitted } from './audioActions';
import { ASSIGN_TASK_PENDING, COMPLETE_REVIEW_PENDING, COMPLETE_TRANSCRIPTION_PENDING, 
    FETCH_TASKS, FETCH_TRANSCRIPTION, SELECT_POPUP_TASK, SELECT_PROJECT, SELECT_TASK,  UNASSIGN_TASK_PENDING,
    UPDATE_TASK, WRITE_FULFILLED, WRITE_PENDING } from './types';
import { saveUserSetting } from './userActions';

export const assignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: ASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Assigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
        .then(dispatch(setSubmitted(false)))
}

export const unAssignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: UNASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Unassigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const completeTranscription = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: COMPLETE_TRANSCRIPTION_PENDING});
    Axios.put('/api/TaskEvent?action=TranscribeEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const completeReview = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: COMPLETE_REVIEW_PENDING});
    Axios.put('/api/TaskEvent?action=ReviewEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(uploadTranscription(taskid, userid)))
}

export const uploadTranscription = (taskid: string, userid: string) => (dispatch: any) => {
    Axios.put('/api/TaskEvent?action=Upload&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const writeTranscription = (taskid: string, length: number, lang: string, dir: string, data: any) => (dispatch: any) => {
    dispatch({type: WRITE_PENDING});
    Axios.put('/api/WriteTranscription?task=' + taskid + "&length=" + length.toString() + "&lang=" + lang + "&dir=" + dir, data)
        .then(dispatch({type: WRITE_FULFILLED}))
}

export const fetchTasks = (username: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?user=' + username)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
        })
}

export const fetchTasksOfProject = (projectname: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?project=' + projectname)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
            dispatch({
                payload: projectname,
                type: SELECT_PROJECT
            })
        })
    }

export function selectProject(id: string): any{
    return {
        payload: id,
        type: SELECT_PROJECT
    }
}

export const  selectTask = (user: string, id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_TASK
    })
    dispatch(fetchTranscription(id));
    dispatch(saveUserSetting(user, "lastTask", id))
    dispatch(setSubmitted(false))
}

export const  selectPopupTask = (id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_POPUP_TASK
    })
}

export const fetchTranscription = (taskid: string) => (dispatch: any) => {
    const part = taskid.split('.');
    Axios.get('/api/audio/' + part[0] + '.transcription').
    then(transcription => {
        dispatch({
            payload: transcription,
            type: FETCH_TRANSCRIPTION
        });
    });
}

export const updateTask = (task: string, project: string, query: string, data: object) => (dispatch: any) => {
    dispatch({type: UPDATE_TASK});
    Axios.put('/api/UpdateTask?task=' + task + '&project=' + project + query, data)
        .then(dispatch(fetchTasksOfProject(project)))
}
