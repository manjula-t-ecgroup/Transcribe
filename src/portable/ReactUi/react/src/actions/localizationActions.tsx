import Axios from 'axios';
import { log } from '../actions/logAction';
import { FETCH_LOCALIZATION } from './types';


export const fetchLocalization = () => (dispatch: any) => {
    Axios.get('/localization/strings.json')
        .then(strings => {
            dispatch({
                payload: strings,
                type: FETCH_LOCALIZATION
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " fetch localization"))
        });
}
