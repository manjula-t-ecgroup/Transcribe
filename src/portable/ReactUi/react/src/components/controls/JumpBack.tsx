import * as React from 'react';
import * as actions from '../../actions/audioActions';
import './JumpBack.css';

interface IProps {
    jump?: number;
    jumpChange: typeof actions.jumpChange;
};

class JumpBack extends React.Component<IProps, object> {
    public render() {
        return (
            <div className="JumpBack" onClick={this.props.jumpChange.bind(this, -2)}>
               {"\u00AB"}
            </div>
            )
    }
};

export default JumpBack;
