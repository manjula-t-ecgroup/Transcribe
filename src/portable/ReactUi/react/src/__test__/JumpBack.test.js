import React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import * as actions from '../actions/audioActions';
import JumpBack from '../components/controls/JumpBack';
import { JSDOM } from 'jsdom';

configure({ adapter: new Adapter() })

// Snapshot for JumpBack
describe('>>>Control: JumpBack --- Snapshot',()=>{
    const minProps = {
        jumpChange: actions.jumpChange,
        selectedUser: "admin",
        users: [{username:{id:"admin"},hotkey:[{id:"back", text:"F2"}]}],
    };

    it('+++capturing Snapshot of JumpBack', () => {
        const { document } = (new JSDOM(<JumpBack {...minProps}/>)).window
        const renderedValue =  renderer.create(document.querySelector('div')).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: JumpBack', () => {
    let wrapper;
    const minProps = {
        jumpChange: actions.jumpChange,
        selectedUser: undefined,
        users: [],
    };

    beforeEach(()=>{
        wrapper = shallow(<JumpBack {...minProps} />)
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});
