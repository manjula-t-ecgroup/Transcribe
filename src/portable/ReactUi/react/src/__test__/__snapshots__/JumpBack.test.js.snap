// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: JumpBack --- Snapshot +++capturing Snapshot of JumpBack 1`] = `null`;
